core/src/com/unciv/ui/components/widgets/HealthBarV0.kt

```diff
@@ -0,0 +1,317 @@
+            (background?.run { leftWidth + rightWidth } ?: 0f)
+        internal fun calcMinHeight() = minBarHeight + 2 * padVertical +
+            (background?.run { bottomHeight + topHeight } ?: 0f)
+    }
+
+    init {
+        if (segmentCount < 2)
+            throw IllegalArgumentException("Less than two segments are nonsense")
+        if (minValue >= maxValue)
+            throw IllegalArgumentException("maxValue must be > minValue")
+    }
+
+    //region internal Fields
+    private val values = FloatArray(segmentCount - 1) { 0f }
+
+    private val drawColor = Color()
+    private val segmentColor = Color()
+    private val drawSegment = Rectangle()
+    private val whiteDot = ImageGetter.getDrawable(null)  // == getWhiteDot().drawable
+
+    private var flashingAction: FlashingAction? = null
+    private var flashingValue = 1f
+
+    private var label: Label? = null
+    //endregion
+
+    operator fun get(index: Int) = values[index]
+    operator fun set(index: Int, value: Float) {
+        values[index] = value.coerceIn(minValue, maxValue)
+        invalidate()
+    }
+
+    /** Shortcut to first value */
+    var value: Float
+        get() = values[0]
+        set(value) { values[0] = value }
+
+    /** Set [style] value [flashingSegment][HealthBarStyle.flashingSegment] **before** turning this on */
+    var allowAnimations = false
+        set(value) {
+            updateAnimations(value)
+            field = value
+        }
+
+    /** Special case for the UnitGroup - one segment colored by value and 1f black border */
+    fun setupForUnitOrCity(current: Int) {
+        style.background = BaseScreen.skinStrings.getUiBackground("General/HealthBar", tintColor = Color.BLACK)
+        style.padVertical = 1f
+        style.padHorizontal = 1.2f
+        values[0] = current.toFloat()
+        style.colors[0] = when {
+            current <= 33 -> Color.RED
+            current <= 66 -> Color.ORANGE
+            else -> Color.GREEN
+        }
+    }
+
+    /** Sets all values.
+     *
+     *  If newValues.size is less than segmentCount-1, only a subset of the values will be updated.
+     *  If newValues.size is greater than segmentCount-1, additional values will be ignored.
+     */
+    fun setValues(newValues: FloatArray) {
+        for (i in newValues.indices) newValues[i] = newValues[i].coerceIn(minValue, maxValue)
+        if (allowAnimations && style.animateDuration > 0f)
+            addAction(AnimateAction(newValues))
+        else
+            newValues.copyInto(values, 0, 0, values.size.coerceAtMost(newValues.size))
+        invalidate()
+    }
+
+    @JvmName("setValuesVarArg")
+    fun setValues(vararg newValues: Float) = setValues(newValues)
+
+    fun setValues(vararg newValues: Int) {
+        val floatValues = FloatArray(newValues.size)
+        for (i in newValues.indices) floatValues[i] = newValues[i].toFloat()
+        setValues(floatValues)
+    }
```

Comment:
Part of clarifying values is clarifying this as well. Maybe values should be a separate class with all the functions it has.